subsys/caf/modules/buttons.c

```diff
@@ -46,17 +46,50 @@ enum state {
 	STATE_SUSPENDING
 };
 
-static const struct device * const gpio_devs[] = {
-	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio0)),
-	DEVICE_DT_GET_OR_NULL(DT_NODELABEL(gpio1)),
+struct gpio_dev {
+	const struct device * const dev;
+	const uint8_t port;
+	const bool has_gpiote;
+	struct gpio_callback gpio_cb;
+};
+
+#define DT_DRV_COMPAT nordic_nrf_gpio
+#define _GPIO_PORT_VALIDATE(id)							\
+	BUILD_ASSERT(DT_INST_PROP(id, port) <= UINT8_MAX);
+#define _GPIO_INSTANCE_GET(id)							\
+	{									\
+		.dev = DEVICE_DT_GET(DT_DRV_INST(id)),				\
+		.port = DT_INST_PROP(id, port),					\
+		.has_gpiote = DT_INST_NODE_HAS_PROP(id, gpiote_instance),	\
+	},
+
+DT_INST_FOREACH_STATUS_OKAY(_GPIO_PORT_VALIDATE)
+static struct gpio_dev gpio_devs[] = {
+	DT_INST_FOREACH_STATUS_OKAY(_GPIO_INSTANCE_GET)
 };
-static struct gpio_callback gpio_cb[ARRAY_SIZE(gpio_devs)];
 static struct k_work_delayable matrix_scan;
 static struct k_work_delayable button_pressed;
 static enum state state;
 static atomic_t system_power_off = ATOMIC_INIT(false);
 
 
+static ssize_t get_gpio_idx(uint8_t port)
+{
+	for (ssize_t i = 0; i < ARRAY_SIZE(gpio_devs); i++) {
+		if (gpio_devs[i].port == port) {
+			return i;
+		}
+	}
+
+	__ASSERT_NO_MSG(false);
+	return -EINVAL;
```

Comment:
`ssize_t` is a signed integer type capable of storing values al least in the range [-1, SSIZE_MAX]. We need to either update value returned here to `-1` or change type of the returned value.